watertap/examples/flowsheets/RO_with_energy_recovery/parameter_sweep_input_parser.py

```diff
@@ -0,0 +1,310 @@
+
+    """
+
+    input_dict = _yaml_to_dict(yaml_filename)
+
+    sweep_params = {}
+
+    for param, values in input_dict.items():
+
+
+        # Split the dot-separated object path into a list of
+        # separate objects and attributes
+        path = _split_pyomo_path(values['model_value'])
+
+        # Follow the attribute path to the correct nested
+        # property, should work with indexing!
+        child = _return_child_property(m, path)
+
+        if values['type'] == 'LinearSample':
+            sweep_params[param] = LinearSample(child,
+                                                values['lower_limit'],
+                                                values['upper_limit'],
+                                                values['num_samples'])
+
+        elif values['type'] == 'UniformSample':
+            sweep_params[param] = UniformSample(child,
+                                                values['lower_limit'],
+                                                values['upper_limit'])
+
+        elif values['type'] == 'NormalSample':
+            sweep_params[param] = NormalSample(child,
+                                               values['mean'],
+                                               values['std'])
+
+        elif values['type'] == 'LatinHypercubeSample':
+            sweep_params[param] = LatinHypercubeSample(child,
+                                                       values['lower_limit'],
+                                                       values['upper_limit'])
+
+    return sweep_params
+
+
+
+def set_defaults_from_yaml(m, yaml_filename, verbose=False, do_nothing=False):
+    """ Sets default model values using values stored in a yaml file
+
+    This function reads a yaml file with the structure::
+
+        m.fs.path.to.attribute_1: 0.123
+        m.fs.path.to.attribute_2: 1.234
+        ...
+
+    and uses the (key, default_value) pairs to set default values
+    for the attributes in model ``m``.
+
+    Args:
+        m (pyomo model):
+            The flowsheet containing the model to set default values for
+        yaml_filename (str):
+            The path to the yaml file.
+
+    Returns:
+        N/A
+
+    """
+
+    input_dict = _yaml_to_dict(yaml_filename)
+
+    fail_ct = 0
+
+    for key, default_value in input_dict.items():
+        # Split the dot-separated object path into a list of
+        # separate objects and attributes
+        path = _split_pyomo_path(key)
+
+        # Follow the attribute path to the correct nested
+        # property, should work with indexing!
+        child = _return_child_property(m, path)
+
+        old_value = value(child)
+
+        if do_nothing:
+            new_value = old_value
+
+        else:
+            new_value = default_value
+
+        if verbose:
+            print('Property: %s' % (key))
+            print('New Value: %e, Old Value: %e' % (new_value, old_value))
```

Comment:
Could we use a logger at the DEBUG level here?

Reply:
Would the suggested fix work, I haven't worked with the logger before so this is copied from a similar use case and may not be the best approach:
```python
import idaes.logger as idaeslog
...
_log = idaeslog.getLogger(__name__)
...
if _log.isEnabledFor(idaeslog.DEBUG):
    _log.debug(f'Property: {key}')
    _log.debug(f'New Value: {new_value:.6e}, Old Value: {old_value:.6e}')
```

Reply:
Added the behavior shown above.